### ISSUE TITLE
[Morph/production] RunWorker::NoRemainingSlotsError: RunWorker::NoRemainingSlotsError

### ISSUE THREAD

**Issue by mlandauer** [openaustralia/morph#1225]
## Backtrace

    line 18 of [PROJECT_ROOT]/app/workers/run_worker.rb: perform

[View full backtrace and more info at honeybadger.io](https://app.honeybadger.io/projects/35868/faults/33985946)

**Comment by jamezpolley** [comment]
Currently, we have 10 containers finished 9 or more days ago but with "Scraper running: yes"

The oldest of these seems to be failing trying to import data into mysql:

````
Excon::Error::Socket: Mysql2::Error: Incorrect string value: '\xF0\x9F\x92\x8E \x0A' for column 'text' at row 1: INSERT INTO `log_lines` (`run_id`, `timestamp`, `stream`, `text`, `created_at`, `updated...
````

**Comment by jamezpolley** [comment]
the 4 oldest stuck jobs match this pattern - unicode characters in log entries.

I've run `docker container rm` on those 4 containers as I couldn't see any other way to resolve this.

`RAILS_ENV=production bundle exec rake app:docker:delete_stopped_containers` cleaned up 4 more containers.

**Comment by jamezpolley** [comment]
Make that 5 scrapers stuck on unicode input:

````
Mysql2::Error: Incorrect string value: '\xF0\xA4\xA7\x9F) ...' for column 'text' at row 1: INSERT INTO `log_lines` (`run_id`, `timestamp`, `stream`, `text`, `created_at`, `updated_at`) VALUES (2245704, '2019-07-03 04:07:06.329594', 'stdout', '第27庭 8字樓 Court No.27 (8/F)-|	|-郭偉健 法官 H.H. Judge W.K. Kwok-|	|-上午 09:30 am 法庭聆訊 Open Court-|	|-刑事案件 DCCC 83/2019-|	|-HKSAR (香港特別行政區) v. A1:WONG PUI TING HEIDI (黃佩𤧟) A2:PALPAL LATOC JO AN EVERA A3:GALLEGO JEANETTE VILLAFLORES-|	|-提訊 Mention-|	|-律政司 Department of Justice 何敦律師行 Haldanes 布高江律師行 Boase, Cohen & Collins\n', '2019-07-12 04:31:43', '2019-07-12 04:31:43') (ActiveRecord::StatementInvalid)
````

**Comment by jamezpolley** [comment]
#601 seems to be the root cause of this issue.

**Comment by jamezpolley** [comment]
I'm closing this as I think the work to fix this is encapsulated in #601 already

**Comment by jamezpolley** [comment]
First step was to fix queue run inconsistencies

````
deploy@morph:/var/www/current$ RAILS_ENV=production bundle exec rake app:emergency:show_queue_run_inconsistencies                                                                                                                                                 Getting information from sidekiq queue...
Getting information about current containers...
The following runs do not have jobs on the queue:
[2241906, 2242196, 2242267, 2242269, 2242330, 2242426, 2242453, 2242456, 2257973, 2258774, 2258779, 2258794]
Unfinished runs attached to scrapers that do not have jobs on the queue:
[2241906, 2242196, 2242267, 2242269, 2242330, 2242426, 2242453, 2242456, 2242459, 2257973, 2258774, 2258779, 2258794]
deploy@morph:/var/www/current$ RAILS_ENV=production bundle exec rake app:emergency:fix_queue_run_inconsistencies                                                                                                                                                  Putting the following runs back on the queue:
[2241906, 2242196, 2242267, 2242269, 2242330, 2242426, 2242453, 2242456, 2242459, 2257973, 2258774, 2258779, 2258794]
````

That didn't immediately fix the problem, as there were not enough slots to tackle those jobs on the queue. Cleaning up other stuck containers freed up slots so that those retries could be processed, which freed up more slots.
